### PR TITLE
[WIP] Newapi

### PR DESCRIPTION
**Pull request by repejota**
* [x] ERNEST-1581 : Deprecate datacenters-data-microservice
* [x] ERNEST-1582 : Deprecate clients-data-microservice
* [x] ERNEST-1583 : Depcrecate users-data-microservice
* [x] ERNEST-1585 : Deprecate services-data-microservice
* [x] ERNEST-1576 : Deploy new group-store
* [x] ERNEST-1565 : Deploy new service-store
* [x] ERNEST-1574 : Deploy new datacenter-store
* [x] ERNEST-1577 : Deploy new user-store
* [x] ERNEST-1586 : Deploy new api gateway
* [ ] Check everything is working after development

### CODE REVIEW

File: .rubocop.yml [hunk 1]
```diff
@@ -1,2 +1,5 @@
 Metrics/LineLength:
   Enabled: false
+
+Style/BlockComments:
```
Comment (adriacidre) [inline .rubocop.yml]: We shouldn't have any comments on the source code, we have git history instead :)

File: attributes/ernest.rb [hunk 2]
```diff
@@ -42,8 +44,13 @@
                                            'firewall-updater-vcloud-connector', 'nat-updater-vcloud-connector',
                                            'router-deleter-vcloud-connector', 'network-deleter-vcloud-connector']
 
+=begin
```
Comment (adriacidre) [inline attributes/ernest.rb]: you've removed all stores from ruby-based deploys but you ddin't added them to to the gpb? they are not going to be deployed...

File: recipes/data_microservices.rb [hunk 3]
```diff
@@ -54,6 +54,7 @@
   end
 end
 
+=begin
```
Comment (adriacidre) [inline recipes/data_microservices.rb]: just remove commented code if no needed

File: templates/groups.sql.erb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+CREATE table IF NOT EXISTS groups (
```
Comment (adriacidre) [inline templates/groups.sql.erb]: The export of the generated groups table looks like this:
```
CREATE TABLE groups (
    id integer NOT NULL,
    group_id integer,
    name text,
    created_at timestamp with time zone,
    updated_at timestamp with time zone,
    deleted_at timestamp with time zone
);

ALTER TABLE ONLY groups ALTER COLUMN id SET DEFAULT nextval('groups_id_seq'::regclass);
ALTER TABLE ONLY groups
    ADD CONSTRAINT groups_pkey PRIMARY KEY (id);
CREATE INDEX idx_groups_deleted_at ON groups USING btree (deleted_at);
CREATE UNIQUE INDEX idx_per_group ON groups USING btree (group_id, name);
```
Additionally it seems to me group_id is not needed as it's repeated with id, so probably [this](https://github.com/ErnestIO/group-store/blob/master/entity.go#L18) column is not needed?

File: templates/users.sql.erb [hunk 5]
```diff
@@ -1,13 +1,14 @@
...
-  user_id VARCHAR(255) NOT NULL PRIMARY KEY,
-  client_id VARCHAR(255) NOT NULL,
-  user_name VARCHAR(255) NOT NULL,
-  user_email VARCHAR(255) NOT NULL,
-  user_password VARCHAR(255) NOT NULL,
-  user_salt VARCHAR(255) NOT NULL,
-  auth_key VARCHAR(255),
-  user_admin boolean NOT NULL default false,
-  UNIQUE (user_name)
+CREATE TABLE users (
```
Comment (adriacidre) [inline templates/users.sql.erb]: dump schema for users:
```
CREATE TABLE users (
    id integer NOT NULL,
    group_id integer,
    username text,
    password text,
    email text,
    salt text,
    admin boolean,
    created_at timestamp with time zone,
    updated_at timestamp with time zone,
    deleted_at timestamp with time zone
);
ALTER TABLE ONLY users ALTER COLUMN id SET DEFAULT nextval('users_id_seq'::regclass);
ALTER TABLE ONLY users
    ADD CONSTRAINT users_pkey PRIMARY KEY (id);
CREATE INDEX idx_users_deleted_at ON users USING btree (deleted_at);
```

**Comment by adriacidre** [comment]
To have everything centralized in this pull request, could you add a new entry to the config-store? we will need a random string "jwt_token" in order to fill the jwt token used on the api-gateway :-)

It will be used here:
https://github.com/ErnestIO/api-gateway/blob/master/setup.go#L29

**Comment by repejota** [comment]
Ok added the new schema and also added a JWT to secrets.rb.

**Pull request merged**